app/forms/school_search_form.rb

```diff
@@ -6,12 +6,20 @@ class SchoolSearchForm
   attr_accessor :school_name, :location, :search_distance, :search_distance_unit, :characteristics, :partnership
 
   def find_schools(page)
-    if partnership&.include? "partnered_with_another_provider"
-      School.where(id: Partnership.select(:school_id).map(&:school_id)).page(page)
-    else
-      School.where("lower(name) LIKE ? OR lower(urn) LIKE ?", "%#{(school_name || '').downcase}%", "%#{(school_name || '').downcase}%").includes(
-        :network, :lead_provider
-      ).page(page)
-    end
+    schools = School.where("lower(name) LIKE ? OR
+                            lower(urn) LIKE ?",
+                           "%#{(school_name || '').downcase}%",
+                           "%#{(school_name || '').downcase}%")
+    .includes(:network, :lead_provider)
+
+    schools = schools.where(id: Partnership.pluck(:school_id)) if filter_by_partnership_status
```

Comment:
I think the purpose of the filter is, when viewed by lead provider XXX:
- checked: "show schools that have been partnered with some other lead provider than XXX" 
- unchecked: "only show schools partnered with XXX or with nobody"

What this does, when viewed by lead provider XXX:
- checked: "show schools that have been partnered with some lead provider" 
- unchecked: "only show schools partnered with with nobody"

I think what it does now is actually reasonably useful, but we might want to rename the label and double check it with Isabelle, cause it feels like a product decision.

Reply:
ahhh ok, i understood the use of the filter as what i've done but can understand why the other purpose would be used. I'll speak with Isabelle about this part of the filters intended use.

Reply:
I had a chat with Alaric and he said that the filter should work as it has been. I've made changes to reflect that its all schools that are partnered